waku/persistence/store.go

```diff
@@ -354,34 +331,69 @@ func (d *DBStore) Query(query *pb.HistoryQuery) (*pb.Index, []StoredMessage, err
 			if query.PagingInfo.Direction == pb.PagingInfo_BACKWARD {
 				eqOp = "<"
 			}
-			paramCnt++
-			conditions = append(conditions, fmt.Sprintf("id %s $%d", eqOp, paramCnt))
+			*paramCnt++
+			conditions = append(conditions, fmt.Sprintf("id %s $%d", eqOp, *paramCnt))
 
 			parameters = append(parameters, cursorDBKey.Bytes())
 		} else {
 			return nil, nil, ErrInvalidCursor
 		}
 	}
 
+	handleTimeParam := func(time int64, op string) {
+		*paramCnt++
+		conditions = append(conditions, fmt.Sprintf("id %s $%d", op, *paramCnt))
+		timeDBKey := NewDBKey(uint64(time), uint64(time), "", []byte{})
+		parameters = append(parameters, timeDBKey.Bytes())
+	}
+
 	if query.StartTime != 0 {
 		if !usesCursor || query.PagingInfo.Direction == pb.PagingInfo_BACKWARD {
-			paramCnt++
-			conditions = append(conditions, fmt.Sprintf("id >= $%d", paramCnt))
-			startTimeDBKey := NewDBKey(uint64(query.StartTime), uint64(query.StartTime), "", []byte{})
-			parameters = append(parameters, startTimeDBKey.Bytes())
+			handleTimeParam(query.StartTime, ">=")
 		}
-
 	}
 
 	if query.EndTime != 0 {
 		if !usesCursor || query.PagingInfo.Direction == pb.PagingInfo_FORWARD {
-			paramCnt++
-			conditions = append(conditions, fmt.Sprintf("id <= $%d", paramCnt))
-			endTimeDBKey := NewDBKey(uint64(query.EndTime), uint64(query.EndTime), "", []byte{})
-			parameters = append(parameters, endTimeDBKey.Bytes())
+			handleTimeParam(query.EndTime, "<=")
 		}
 	}
+	return conditions, parameters, nil
+}
 
+func (d *DBStore) prepareQuerySQL(query *pb.HistoryQuery) (string, []interface{}, error) {
+	sqlQuery := `SELECT id, receiverTimestamp, senderTimestamp, contentTopic, pubsubTopic, payload, version 
+	FROM message 
+	%s
+	ORDER BY senderTimestamp %s, id %s, pubsubTopic %s, receiverTimestamp %s `
+
+	var conditions []string
+	var parameters []interface{}
+	//parameters := make([]interface{}, 0) //Allocating size of 1 as pageSize is gauranteed param to be present.
```

Comment:
```suggestion
```

Reply:
Was running into some SQL issue when i changed it to slice. 
But now i think that might have been probably due to other issues i fixed later. Will change it and test.